src/models/user.ts

```diff
@@ -0,0 +1,46 @@
+import { BaseUser, RegisterUser, User, ResponseUser } from "../types/user";
+import dbService from "../services/db";
+import { OkPacket, RowDataPacket } from "mysql2";
+
+const create = (user: RegisterUser) : Promise<number | null> => new Promise((resolve, reject) => {
+  const queryString = `INSERT INTO User (email, name, password, role) VALUES (?, ?, ?, ?)`;
+
+  dbService.query(
+    queryString,
+    [user.email, user.name, user.password, user.role],
+    (err, result) => {
+      // console.log("createV2", err, result)
+      if (err !== null) return reject(err);
+
+      const insertId = (<OkPacket> result).insertId;
+      resolve(insertId);
+    }
+  );
+});
+
+const findOne = (email: string) : Promise<User | null> => new Promise((resolve, reject) => {
+  const queryString = `SELECT * from User WHERE email=?`;
+
+  dbService.query(queryString, email, (err, result) => {
+    // console.log("findOneV2", err, result)
+    if (err !== null) return reject(err);
+
+    if ((<RowDataPacket>result).length === 0) return resolve(null);
```

Comment:
lebih rekomen result as RowDataPacket dibanding <RowDataPacket>result, karena tanda <> lebih indentik sama generic

Reply:
generic2 yang lain better saya ubah ke as ya kak?

Reply:
Enggak, maksudnya kalo mau pake tanda <>, dipakenya digeneric aja contoh:
Promise<string> ini generic,
kalo <string>result atau result as string ini type casting
Jadi kalo typecasting pake as
kalo generic pake <>

Reply:
ok got it kak, masalah penulisan aja ya biar lebih gampang bedain yang generic atau bukan